core/DataTable/Filter/Sort.php

```diff
@@ -262,8 +255,14 @@ private function sort(DataTable $table, $functionCallback)
 
         // get column value and label only once for performance tweak
         $values = array();
-        foreach ($rows as $key => $row) {
-            $values[$key] = array($this->getColumnValue($row), $row->getColumn('label'));
+        if ($functionCallback === 'numberSort') {
+            foreach ($rows as $key => $row) {
+                $values[$key] = array($this->getColumnValue($row), $row->getColumn('label'));
+            }
+        } else {
+            foreach ($rows as $key => $row) {
```

Comment:
Only `numberSort` actually uses a secondary column (`label`). For all other sort callbacks we do not need the secondary column so we can simply pass the value directly which will make sorting by label faster.